src/components/Rockets.jsx

```diff
@@ -16,8 +16,8 @@ const Rockets = () => {
       .catch(() => {});
   }, [dispatch]);
 
-  const handleReserve = (rocketId) => {
-    dispatch(reserveRocket(rocketId));
+  const handleReserve = (rocketId, rocketTitle) => {
+    dispatch(reserveRocket(rocketId, rocketTitle));
   };
 
   const handleCancelReservation = (rocketId) => {
```

Comment:
When a rocket is reserved, and the user subsequently navigates from the Rockets page to the My Profile page and back, the previously reserved rockets, along with the 'reserved' button, are no longer displayed. This issue arises because the Rockets component is rendering the rocket data every time the user visits the page.


https://github.com/lily4178993/space-traveller/assets/101261047/57cc21bf-24d7-4391-ae1a-d5314524a7f8

To address this issue, a solution is to introduce a conditional statement to verify the length of the rocket data and only dispatch the data when it is not empty. Below is the updated code:

```js
useEffect(() => {
// Insert this conditional statement in your Rockets component
    if (rocketData.length === 0) {
      fetch('https://api.spacexdata.com/v3/rockets')
        .then((response) => response.json())
        .then((data) => {
// Dispatch the action to fetch rockets only when the data is not present
          dispatch(fetchRockets(data));
        })
        .catch(() => {});
    }
  }, [dispatch, rocketData.length]);
```